src/ring_theory/graded_algebra/homogeneous_ideal.lean

```diff
@@ -471,3 +471,104 @@ lemma ideal.homogeneous_hull_eq_Inf (I : ideal A) :
 eq.symm $ is_glb.Inf_eq $ (ideal.homogeneous_hull.gc 𝒜).is_least_l.is_glb
 
 end galois_connection
+
+section irrelevant_ideal
+
+open graded_algebra finset set_like.graded_monoid
+
+variables [comm_semiring R] [semiring A]
+variables [algebra R A] [decidable_eq ι]
+variables [canonically_ordered_add_monoid ι] [covariant_class ι ι has_add.add has_lt.lt]
+variables (𝒜 : ι → submodule R A) [graded_algebra 𝒜]
+
+/--
+If `A` is graded by a canonically ordered add monoid, then the projection map `x ↦ x₀` is a ring
+homomorphism.
+-/
+def graded_algebra.proj_zero_ring_hom : A →+* 𝒜 0 :=
+{ to_fun := λ a, ⟨proj 𝒜 0 a, submodule.coe_mem _⟩,
```

Comment:
```suggestion
{ to_fun := λ a, decompose 𝒜 a 0,
```

Reply:
I just tried to use `decompose` and the proof is not very smooth because,
`(decompose 𝒜 x 0 * decompose 𝒜 y 0).1` and `(decompose 𝒜 x 0).1 * (decompose 𝒜 y 0).1` aren't definitionally equal.

Reply:
done

Reply:
I changed to `λ a, decompose 𝒜 a 0` and used the semiring instance in `direct_sum/ring.lean`. This broke the proof, in particular because `(decompose 𝒜 x 0 * decompose 𝒜 y 0).val` is not definitionally equal to `(decompose 𝒜 x 0).1 * (decompose 𝒜 y 0).1`. How is multiplication in `grade_zero.semiring` defined.

Reply:
I changed back to `A→+* A`, because it actually makes things easier to prove and with less imports. Is this ok?

Reply:
Yes that's fine, I experimented with this and was worried it might turn out annoying